scipy/stats/tests/test_variation.py

```diff
@@ -0,0 +1,134 @@
+    """
+
+    def test_ddof(self):
+        x = np.arange(9.0)
+        assert_allclose(variation(x, ddof=1), np.sqrt(60/8)/4)
+
+    @pytest.mark.parametrize('sgn', [1, -1])
+    def test_sign(self, sgn):
+        x = np.array([1, 2, 3, 4, 5])
+        v = variation(sgn*x)
+        expected = sgn*np.sqrt(2)/3
+        assert_allclose(v, expected, rtol=1e-10)
+
+    def test_scalar(self):
+        # A scalar is treated like a 1-d sequence with length 1.
+        assert_equal(variation(4.0), 0.0)
+
+    @pytest.mark.parametrize('nan_policy, expected',
+                             [('propagate', np.nan),
+                              ('omit', np.sqrt(20/3)/4)])
+    def test_variation_nan(self, nan_policy, expected):
+        x = np.arange(10.)
+        x[9] = np.nan
+        assert_allclose(variation(x, nan_policy=nan_policy), expected)
+
+    def test_nan_policy_raise(self):
+        x = np.array([1.0, 2.0, np.nan, 3.0])
+        with pytest.raises(ValueError, match='input contains nan'):
+            variation(x, nan_policy='raise')
+
+    def test_bad_nan_polciy(self):
+        with pytest.raises(ValueError, match='must be one of'):
+            variation([1, 2, 3], nan_policy='foobar')
+
+    def test_keepdims(self):
+        x = np.arange(10).reshape(2, 5)
+        y = variation(x, axis=1, keepdims=True)
+        expected = np.array([[np.sqrt(2)/2],
+                             [np.sqrt(2)/7]])
+        assert_allclose(y, expected)
+
+    def test_propagate_nan(self):
+        # Check that the shape of the result is the same for inputs
+        # with and without nans, cf gh-5817
+        a = np.arange(8).reshape(2, -1).astype(float)
+        a[1, 0] = np.nan
+        v = variation(a, axis=1, nan_policy="propagate")
+        assert_allclose(v, [np.sqrt(5/4)/1.5, np.nan], atol=1e-15)
+
+    def test_axis_none(self):
+        # Check that `variation` computes the result on the flattened
+        # input when axis is None.
+        y = variation([[0, 1], [2, 3]], axis=None)
+        assert_allclose(y, np.sqrt(5/4)/1.5)
+
+    def test_bad_axis(self):
+        # Check that an invalid axis raises np.AxisError.
+        x = np.array([[1, 2, 3], [4, 5, 6]])
+        with pytest.raises(np.AxisError):
+            variation(x, axis=10)
+
+    def test_mean_zero(self):
+        # Check that `variation` returns inf for a sequence that is not
+        # identically zero but whose mean is zero.
+        x = np.array([10, -3, 1, -4, -4])
+        y = variation(x)
+        assert_equal(y, np.inf)
+
+        x2 = np.array([x, -10*x])
+        y2 = variation(x2, axis=1)
+        assert_equal(y2, [np.inf, np.inf])
+
+    @pytest.mark.parametrize('x', [np.zeros(5), []])
+    def test_all_zero(self, x):
+        # Test some cases where `variation` return nan.
+        y = variation(x)
+        assert_equal(y, np.nan)
+
+    @pytest.mark.parametrize('axis, expected',
+                             [(0, []), (1, [np.nan]*3), (None, np.nan)])
```

Comment:
I'm not sure here so I'll ask, is it worthwhile to add a case w/just `inf`s here too?

Reply:
Do you mean the input is `inf`, or the result is `inf`?

Reply:
> Do you mean the input is `inf`, or the result is `inf`?

I was thinking when the input is `inf`, I'd guess in that case the expected output is `nan` which is why I was asking. If there is also a case with `inf` outputs that makes sense in the context of the test sure but the specifics of that input are not apparent to me at this time.

Reply:
@rlucas7, I'm reviving this PR and catching up on some old comments.

An `inf` in the input will result in the output being `nan`.  The call to `numpy.std` will generate a warning in this case: `RuntimeWarning: invalid value encountered in subtract` that isn't very helpful.  I'll avoid that warning, and add a note that an `inf` in the input results in `nan`.

Reply:
Done.